lib/puppetserver/ca/action/prune.rb

```diff
@@ -187,6 +196,34 @@ def prune_using_certname(crl, key, inventory_file, cadir, certnames)
           end
         end
 
+        def prune_expired (crl, key, inventory_file, cadir)
+          serialnumbers = []
+          signed_dir = "#{cadir}/signed"
+          @logger.debug("Checking inventory file #{inventory_file} for expired entries") if @logger.debug?
+          errors = FileSystem.validate_file_paths(inventory_file)
+          if errors.empty?
+            File.open(inventory_file).each_line do |line|
```

Comment:
If you're open to it, perhaps use the Inventory library added recently?  This will parse it for you and return a hash you can act on, handling any errors that arise.

Reply:
I looked at the library. The serial number check anyway will be done [here.](https://github.com/puppetlabs/puppetserver-ca-cli/pull/108/files#diff-c3becff886d5f2e548cbd5bdf8233fc0156da22cbd17ed00c127baff5c8a98c6R154). With the library it will do that check twice and also the serial number check has to be done against older-serial and new-serial where as it is just a line match here.

Reply:
hmm...that didn't work for me. With that default config, the crl is not getting set correctly. So I am going to leave it as it is.